Fix NSFontChanging changeFont signature mismatch warning in AppDelegate

## Summary
Building SignboardApp emits a Swift warning in `Sources/SignboardApp/AppDelegate.swift` because `changeFont(_:)` nearly matches `NSFontChanging.changeFont(_:)` but the parameter type does not match.

## Current Warning
`instance method 'changeFont' nearly matches optional requirement 'changeFont' of protocol 'NSFontChanging'`

## Location
- `Sources/SignboardApp/AppDelegate.swift:192`

## Why This Matters
- Build output is noisy and can hide actionable warnings.
- The current method signature is ambiguous with the AppKit protocol requirement.

## Proposed Fix
- Update the method signature to align with `NSFontChanging`:
  - from: `@objc func changeFont(_ sender: Any?)`
  - to: `@MainActor @objc func changeFont(_ sender: NSFontManager?)`
- Keep behavior unchanged when `sender` is nil.
- Ensure this still works with the existing font panel action flow.

## Acceptance Criteria
- `swift build` no longer reports the `changeFont` nearly-matches-protocol warning.
- Font change behavior in the app remains unchanged.


## Raw Build Warning (as reported)
```text
Sources/SignboardApp/AppDelegate.swift:192:16: warning: instance method 'changeFont' nearly matches optional requirement 'changeFont' of protocol 'NSFontChanging'
190 |     }
191 |
192 |     @objc func changeFont(_ sender: Any?) {
    |                |- warning: instance method 'changeFont' nearly matches optional requirement 'changeFont' of protocol 'NSFontChanging'
    |                |- note: candidate has non-matching type '(Any?) -> ()'
    |                |- note: move 'changeFont' to an extension to silence this warning
    |                `- note: make 'changeFont' private to silence this warning
193 |         guard let manager = sender as? NSFontManager else {
194 |             return

AppKit.NSFontChanging.changeFont:2:26: note: requirement 'changeFont' declared here
1 | protocol NSFontChanging {
2 | @MainActor optional func changeFont(_ sender: NSFontManager?)}
  |                          `- note: requirement 'changeFont' declared here
3 |
Sources/SignboardApp/AppDelegate.swift:192:16: warning: instance method 'changeFont' nearly matches optional requirement 'changeFont' of protocol 'NSFontChanging'
190 |     }
191 |
192 |     @objc func changeFont(_ sender: Any?) {
    |                |- warning: instance method 'changeFont' nearly matches optional requirement 'changeFont' of protocol 'NSFontChanging'
    |                |- note: candidate has non-matching type '(Any?) -> ()'
    |                |- note: move 'changeFont' to an extension to silence this warning
    |                `- note: make 'changeFont' private to silence this warning
193 |         guard let manager = sender as? NSFontManager else {
194 |             return

AppKit.NSFontChanging.changeFont:2:26: note: requirement 'changeFont' declared here
1 | protocol NSFontChanging {
2 | @MainActor optional func changeFont(_ sender: NSFontManager?)}
  |                          `- note: requirement 'changeFont' declared here
3 |
Sources/SignboardApp/AppDelegate.swift:192:16: warning: instance method 'changeFont' nearly matches optional requirement 'changeFont' of protocol 'NSFontChanging'
190 |     }
191 |
192 |     @objc func changeFont(_ sender: Any?) {
    |                |- warning: instance method 'changeFont' nearly matches optional requirement 'changeFont' of protocol 'NSFontChanging'
    |                |- note: candidate has non-matching type '(Any?) -> ()'
    |                |- note: move 'changeFont' to an extension to silence this warning
    |                `- note: make 'changeFont' private to silence this warning
193 |         guard let manager = sender as? NSFontManager else {
194 |             return

AppKit.NSFontChanging.changeFont:2:26: note: requirement 'changeFont' declared here
1 | protocol NSFontChanging {
2 | @MainActor optional func changeFont(_ sender: NSFontManager?)}
  |                          `- note: requirement 'changeFont' declared here
3 |
```

--- a/Sources/SignboardApp/AppDelegate.swift
+++ b/Sources/SignboardApp/AppDelegate.swift
@@ -189,8 +189,8 @@ final class AppDelegate: NSObject, NSApplicationDelegate, NSFontChanging, NSUser
         return true
     }
 
-    @objc func changeFont(_ sender: Any?) {
-        guard let manager = sender as? NSFontManager else {
+    @MainActor @objc func changeFont(_ sender: NSFontManager?) {
+        guard let manager = sender else {
             return
         }
         let converted = manager.convert(currentFont())

